Mathlib/Data/Polynomial/FieldDivision.lean

```diff
@@ -0,0 +1,571 @@
+#align polynomial.normalize_monic Polynomial.normalize_monic
+
+-- Porting note: previously could not synthesize NormalisationMonoid R[X]
+set_option synthInstance.etaExperiment true in
+theorem map_dvd_map' [Field k] (f : R →+* k) {x y : R[X]} : x.map f ∣ y.map f ↔ x ∣ y :=
+  if H : x = 0 then by rw [H, Polynomial.map_zero, zero_dvd_iff, zero_dvd_iff, map_eq_zero]
+  else by
+    rw [← normalize_dvd_iff, ← @normalize_dvd_iff R[X], normalize_apply, normalize_apply,
+      coe_normUnit_of_ne_zero H, coe_normUnit_of_ne_zero (mt (map_eq_zero f).1 H),
+      leadingCoeff_map, ← map_inv₀ f, ← map_C, ← Polynomial.map_mul,
+      map_dvd_map _ f.injective (monic_mul_leadingCoeff_inv H)]
+#align polynomial.map_dvd_map' Polynomial.map_dvd_map'
+
+-- Porting note: previously could not synthesize NormalisationMonoid R[X]
+set_option synthInstance.etaExperiment true in
+theorem degree_normalize : degree (normalize p) = degree p := by simp
+#align polynomial.degree_normalize Polynomial.degree_normalize
+
+-- Porting note: previously could not synthesize NormalisationMonoid R[X]
+set_option synthInstance.etaExperiment true in
+theorem prime_of_degree_eq_one (hp1 : degree p = 1) : Prime p :=
+  have : Prime (normalize p) :=
+    Monic.prime_of_degree_eq_one (hp1 ▸ degree_normalize)
+      (monic_normalize fun hp0 => absurd hp1 (hp0.symm ▸ by simp))
+  (normalize_associated _).prime this
+#align polynomial.prime_of_degree_eq_one Polynomial.prime_of_degree_eq_one
+
+theorem irreducible_of_degree_eq_one (hp1 : degree p = 1) : Irreducible p :=
+  (prime_of_degree_eq_one hp1).irreducible
+#align polynomial.irreducible_of_degree_eq_one Polynomial.irreducible_of_degree_eq_one
+
+theorem not_irreducible_c (x : R) : ¬Irreducible (C x) :=
+  if H : x = 0 then by
+    rw [H, C_0]
+    exact not_irreducible_zero
+  else fun hx => Irreducible.not_unit hx <| isUnit_C.2 <| isUnit_iff_ne_zero.2 H
+set_option linter.uppercaseLean3 false in
+#align polynomial.not_irreducible_C Polynomial.not_irreducible_c
+
+theorem degree_pos_of_irreducible (hp : Irreducible p) : 0 < p.degree :=
+  lt_of_not_ge fun hp0 =>
+    have := eq_C_of_degree_le_zero hp0
+    not_irreducible_c (p.coeff 0) <| this ▸ hp
+#align polynomial.degree_pos_of_irreducible Polynomial.degree_pos_of_irreducible
+
+/- Porting note: factored out a have statement from isCoprime_of_is_root_of_eval_derivative_ne_zero 
+into multiple decls because the original proof was timing out -/
+theorem id_sub_modByMonic_eq_X_sub_C_prod_divByMonic {K : Type _} [Field K] (f : K[X]) (a : K) :
+    (X - C a) * (f /ₘ (X - C a)) = f - f %ₘ (X - C a) := by
+  rw [eq_sub_iff_add_eq, ← eq_sub_iff_add_eq', modByMonic_eq_sub_mul_div]
+  exact monic_X_sub_C a
+
+/- Porting note: factored out a have statement from isCoprime_of_is_root_of_eval_derivative_ne_zero 
+because the original proof was timing out -/
+theorem derivative_eq_divByMonic_add_X_sub_C_prod_derivative_divByMonic 
```

Comment:
this should be called `divByMonic_add_X_Sub_C_mul_derivate_divByMonic_eq_derivative`

Reply:
Done